GenericsHomework.Tests/NodeTests.cs

```diff
@@ -0,0 +1,116 @@
+using Xunit;
+
+namespace GenericsHomework.Tests;
+
+public class NodeTests
+{
+    [Fact]
+    public void Constructor_CorrectInput_SetsValues()
+    {
+        Node<int> node = new(5);
+        Assert.Equal(5, node.Data);
+    }
+
+    [Fact]
+    public void ToString_ReturnsCorrect()
+    {
+        Node<int> node = new(1999);
+        string res = node.ToString();
+        Assert.Equal("1999", res);
+    }
+
+    [Fact]
+    public void Append_AddingNode_AddsNode()
+    {
+        Node<int> node = new(1);
+        node.Append(2);
+        Assert.Equal(2, node.Next.Data);
+    }
+
+    [Fact]
+    public void Clear_NodePointsToSelf()
+    {
+        Node<int> node = new(1);
+        node.Clear();
+        Assert.True(object.ReferenceEquals(node, node.Next));
+    }
+
+    [Fact]
+    public void Next_InitiallyPointsToSelf()
+    {
+        Node<int> node = new(1);
+        Assert.True(object.ReferenceEquals(node, node.Next));
+    }
+
+    [Fact]
+    public void Exists_ValueExists_ReturnsTrue()
+    {
+        Node<int> node = new(1);
+        bool exists = node.Exists(1);
+        Assert.True(exists);
+    }
```

Comment:
```suggestion
    [Theory]
    [InlineData("Hi")]
    [InlineData(null!)]
    public void Exists_ValueExists_ReturnsTrue(string data)
    {
        Node<string> node = new(data);
        bool exists = node.Exists(data);
        Assert.True(exists);
    }
```
Here's an example for making the Null and non-null test one test. Exists for non-null and null data is doing the same thing so only one test is necessary in my opinion